depends/patches/qt/qtbase_avoid_qmain.patch

```diff
@@ -0,0 +1,17 @@
+Avoid defining QT_NEEDS_QMAIN macro
+
+This macro is required for linking to the Qt DLL on Windows, and it
+breaks the CONTROL_FLOW security check during Guix build for Windows.
+
```

Comment:
Could you better explain this patch? The description says that using the `QT_NEEDS_QMAIN` macro is required, but using it breaks things, so you're just getting rid of it? So how do things continue to work, if you're removing the required macro?

Reply:
> Could you better explain this patch?

Sure.

> The description says that using the `QT_NEEDS_QMAIN` macro is required...

... "for linking to the Qt DLL on Windows", which is not our use case.

> ... but using it breaks things, so you're just getting rid of it? So how do things continue to work, if you're removing the required macro?

Using static libraries for Windows does not require all Windows-specific stuff with export/import declarations, entry points etc.

Reply:
It'd be good to add all the relevant information and context to the patch. Can you also explain "breaks the CONTROL_FLOW security check during Guix build for Windows."

Reply:
How about this description:
```
Avoid defining QT_NEEDS_QMAIN macro

Qt's QT_NEEDS_QMAIN macro renames our main() function to qMain(),
which breaks our assumptions regarding exported symbols. In particular,
the CONTROL_FLOW security check fails for Windows builds in Guix.

The QT_NEEDS_QMAIN macro is required for linking to the Qt DLLs only,
so we can safely disable it.
```
?

Reply:
This seems more clear.

Reply:
Thanks! Updated.